azurerm/resource_arm_network_watcher_flow_log.go

```diff
@@ -0,0 +1,387 @@
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"network_watcher_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"network_security_group_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"storage_account_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+
+			"retention_policy": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:             schema.TypeBool,
+							Required:         true,
+							DiffSuppressFunc: azureRMSuppressFlowLogRetentionPolicyEnabledDiff,
+						},
+
+						"days": {
+							Type:             schema.TypeInt,
+							Required:         true,
+							DiffSuppressFunc: azureRMSuppressFlowLogRetentionPolicyDaysDiff,
+						},
+					},
+				},
+			},
+
+			"traffic_analytics": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"workspace_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.UUID,
+						},
+
+						"workspace_region": {
+							Type:             schema.TypeString,
+							Required:         true,
+							StateFunc:        azure.NormalizeLocation,
+							DiffSuppressFunc: azure.SuppressLocationDiff,
+						},
+
+						"workspace_resource_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: azure.ValidateResourceIDOrEmpty,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func azureRMSuppressFlowLogRetentionPolicyEnabledDiff(k, old, new string, d *schema.ResourceData) bool {
+	// Ignore if flow log is disabled as the returned flow log configuration
+	// returns default value `false` which may differ from config
+	return old != "" && !d.Get("enabled").(bool)
+}
+
+func azureRMSuppressFlowLogRetentionPolicyDaysDiff(k, old, new string, d *schema.ResourceData) bool {
+	// Ignore if flow log is disabled as the returned flow log configuration
+	// returns default value `0` which may differ from config
+	return old != "" && !d.Get("enabled").(bool)
```

Comment:
Should this checking `days` instead?

Reply:
Nah, it checks `enabled` as when its false the API returns 0 for days